src/instruments.rs

```diff
@@ -1,133 +1,331 @@
 //! interfacing with the `instruments` command line tool
 
 use std::fs;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 use std::process::{Command, Output};
 
 use anyhow::{anyhow, Result};
+use cargo::core::Workspace;
+use semver::Version;
 
-use crate::opt::Opts;
+use crate::opt::AppConfig;
 
-/// Check that `instruments` is in $PATH.
-pub(crate) fn check_existence() -> Result<()> {
-    let path = ["/", "usr", "bin", "instruments"].iter().collect::<PathBuf>();
-    if path.exists() {
-        Ok(())
-    } else {
+/// Holds available templates.
+///
+pub struct TemplateCatalog {
+    standard_templates: Vec<String>,
+    custom_templates: Vec<String>,
+}
+
+impl TemplateCatalog {
+    /// Return a vector of all templates.
+    fn all_templates(&self) -> Vec<&str> {
+        self.standard_templates
+            .iter()
+            .chain(self.custom_templates.iter())
+            .map(|value| value.as_str())
+            .collect::<Vec<_>>()
```

Comment:
this really doesn't matter in this particular case, but as a general observation I would prefer the pattern of returning `impl Iterator<Item=&str>` here, and then if the caller wants a vec they can handle the collecting, but if they didn't need a vec then we let them skip an allocation.

Again this isn't important here, but thought I'd mention it.

Reply:
Thanks, I also tend to favor composability and minimize allocations, but as you point out, here I think it's ok.